src/DIRAC/FrameworkSystem/scripts/dirac_login.py

```diff
@@ -1,237 +1,380 @@
+        Script.registerSwitch("I:", "issuer=", "set issuer.", self.setIssuer)
+        Script.registerSwitch(
+            " ",
+            "use-certificate",
+            "in case you want to generate a proxy using a certificate. By default.",
+            self.useCertificate,
+        )
+        Script.registerSwitch(
+            " ", "use-diracas", "in case you want to authorize with DIRAC Authorization Server.", self.useDIRACAS
+        )
+        Script.registerSwitch("C:", "certificate=", "user certificate location", self.setCertificate)
+        Script.registerSwitch("K:", "key=", "user key location", self.setPrivateKey)
+        Script.registerSwitch(" ", "proxy", "return proxy in case of successful authorization", self.setProxy)
+        Script.registerSwitch(" ", "token", "return tokens in case of successful authorization", self.setToken)
+        Script.registerSwitch(" ", "status", "print user authorization status", self.authStatus)
 
     def doOAuthMagic(self):
         """Magic method with tokens
 
         :return: S_OK()/S_ERROR()
         """
-        tokenFile = getTokenFileLocation(self.tokenLoc)
-
-        if self.info:
-            # Try to get user information
-            Script.enableCS()
-
-            useTokens = DIRAC.gConfig.getValue("/DIRAC/Security/UseTokens", "false").lower() in ("y", "yes", "true")
-            if "DIRAC_USE_ACCESS_TOKEN" in os.environ:
-                useTokens = os.environ.get("DIRAC_USE_ACCESS_TOKEN", "false").lower() in ("y", "yes", "true")
-            if useTokens:
-                gLogger.notice(
-                    "You are currently using access token to access new HTTP DIRAC services."
-                    " To use a proxy instead, do the following:\n",
-                    "export DIRAC_USE_ACCESS_TOKEN=False\n",
-                )
-                result = readTokenFromFile(tokenFile)
-                if result["OK"]:
-                    gLogger.notice(result["Value"].getInfoAsString())
-            else:
-                gLogger.notice(
-                    "You are currently using proxy to access new HTTP DIRAC services."
-                    " To use a access token instead, do the following:\n",
-                    "export DIRAC_USE_ACCESS_TOKEN=True\n",
-                )
-                result = getProxyInfo(self.proxyLoc)
-                if result["OK"]:
-                    gLogger.notice(formatProxyInfoAsString(result["Value"]))
-
-            return result
-
         params = {}
         if self.issuer:
             params["issuer"] = self.issuer
         result = IdProviderFactory().getIdProvider("DIRACCLI", **params)
         if not result["OK"]:
             return result
         idpObj = result["Value"]
-        scope = []
-        if self.group:
-            scope.append("g:%s" % self.group)
-        if self.proxy:
-            scope.append("proxy")
+        if self.group and self.group not in self.scopes:
+            self.scopes.append(f"g:{self.group}")
```

Comment:
I am not familiar with this. For my info, what does "g:group" means?

Reply:
`DIRAC Authorization Server` - OAuth 2.0 Authorization Server.
`g:<group name>` is a parametric scope.

Reply:
When we send a request for authorization, we pass the desired scopes, which will be added to the access token. `DIRAC AS` will not issue its access token, but it can select the appropriate scope to request the target IdP(e.g. IAM).